File: tests/testdata/codegen/assets-archives-pp/python/assets-archives.py
import codeinfra
import codeinfra_aws as aws

site_bucket = aws.s3.Bucket("siteBucket")
test_file_asset = aws.s3.BucketObject("testFileAsset",
    bucket=site_bucket.id,
    source=codeinfra.FileAsset("file.txt"))
test_string_asset = aws.s3.BucketObject("testStringAsset",
    bucket=site_bucket.id,
    source=codeinfra.StringAsset("<h1>File contents</h1>"))
test_remote_asset = aws.s3.BucketObject("testRemoteAsset",
    bucket=site_bucket.id,
    source=codeinfra.RemoteAsset("https://codeinfra.test"))
test_file_archive = aws.lambda_.Function("testFileArchive",
    role=site_bucket.arn,
    code=codeinfra.FileArchive("file.tar.gz"))
test_remote_archive = aws.lambda_.Function("testRemoteArchive",
    role=site_bucket.arn,
    code=codeinfra.RemoteArchive("https://codeinfra.test/foo.tar.gz"))
test_asset_archive = aws.lambda_.Function("testAssetArchive",
    role=site_bucket.arn,
    code=codeinfra.AssetArchive({
        "file.txt": codeinfra.FileAsset("file.txt"),
        "string.txt": codeinfra.StringAsset("<h1>File contents</h1>"),
        "remote.txt": codeinfra.RemoteAsset("https://codeinfra.test"),
        "file.tar": codeinfra.FileArchive("file.tar.gz"),
        "remote.tar": codeinfra.RemoteArchive("https://codeinfra.test/foo.tar.gz"),
        ".nestedDir": codeinfra.AssetArchive({
            "file.txt": codeinfra.FileAsset("file.txt"),
            "string.txt": codeinfra.StringAsset("<h1>File contents</h1>"),
            "remote.txt": codeinfra.RemoteAsset("https://codeinfra.test"),
            "file.tar": codeinfra.FileArchive("file.tar.gz"),
            "remote.tar": codeinfra.RemoteArchive("https://codeinfra.test/foo.tar.gz"),
        }),
    }))

File: tests/testdata/codegen/assets-archives-pp/nodejs/assets-archives.ts
import * as codeinfra from "@codeinfra/codeinfra";
import * as aws from "@codeinfra/aws";

const siteBucket = new aws.s3.Bucket("siteBucket", {});
const testFileAsset = new aws.s3.BucketObject("testFileAsset", {
    bucket: siteBucket.id,
    source: new codeinfra.asset.FileAsset("file.txt"),
});
const testStringAsset = new aws.s3.BucketObject("testStringAsset", {
    bucket: siteBucket.id,
    source: new codeinfra.asset.StringAsset("<h1>File contents</h1>"),
});
const testRemoteAsset = new aws.s3.BucketObject("testRemoteAsset", {
    bucket: siteBucket.id,
    source: new codeinfra.asset.RemoteAsset("https://codeinfra.test"),
});
const testFileArchive = new aws.lambda.Function("testFileArchive", {
    role: siteBucket.arn,
    code: new codeinfra.asset.FileArchive("file.tar.gz"),
});
const testRemoteArchive = new aws.lambda.Function("testRemoteArchive", {
    role: siteBucket.arn,
    code: new codeinfra.asset.RemoteArchive("https://codeinfra.test/foo.tar.gz"),
});
const testAssetArchive = new aws.lambda.Function("testAssetArchive", {
    role: siteBucket.arn,
    code: new codeinfra.asset.AssetArchive({
        "file.txt": new codeinfra.asset.FileAsset("file.txt"),
        "string.txt": new codeinfra.asset.StringAsset("<h1>File contents</h1>"),
        "remote.txt": new codeinfra.asset.RemoteAsset("https://codeinfra.test"),
        "file.tar": new codeinfra.asset.FileArchive("file.tar.gz"),
        "remote.tar": new codeinfra.asset.RemoteArchive("https://codeinfra.test/foo.tar.gz"),
        ".nestedDir": new codeinfra.asset.AssetArchive({
            "file.txt": new codeinfra.asset.FileAsset("file.txt"),
            "string.txt": new codeinfra.asset.StringAsset("<h1>File contents</h1>"),
            "remote.txt": new codeinfra.asset.RemoteAsset("https://codeinfra.test"),
            "file.tar": new codeinfra.asset.FileArchive("file.tar.gz"),
            "remote.tar": new codeinfra.asset.RemoteArchive("https://codeinfra.test/foo.tar.gz"),
        }),
    }),
});
